File: go/src/ipaddress/prefix/prefix32/prefix32.go
package prefix32

// import "./prefix"
import "../../ip_bits"
import "../../prefix"
import "fmt"

func From(my *prefix.Prefix, num uint8) prefix.ResultPrefix {
    return New(num);
}
///  Gives the prefix in IPv4 dotted decimal format,
///  i.e. the canonical netmask we're all used to
///
///    prefix = IPAddress::prefix::Prefix32.new 24
///
///    prefix.to_ip
///      ///  "255.255.255.0"
///
func New(num uint8) prefix.ResultPrefix {
    if num <= 32 {
        ipBits := ip_bits.V4();
        bits := ipBits.Bits;
        tmp := prefix.New_netmask(uint8(num), bits)
        return &prefix.Ok{&prefix.Prefix {
            uint8(num),
            ipBits,
            *tmp,
            From,
            //vt_to_ip_str: _TO_IP_STR,
        }};
    }
    tmp := fmt.Sprintf("Prefix must be in range 0..32, got: %d", num)
    return &prefix.Error{&tmp}
}

File: python/src/ip_address/prefix32.py
from typing import Optional

from .ip_bits import IpBits
from .prefix import Prefix


class Prefix32:
    @staticmethod
    def from_(my: Prefix, num: int) -> Optional[Prefix]:
        return Prefix32.create(num)

    @staticmethod
    def create(num: int) -> Optional[Prefix]:
        if 0 <= num <= 32:
            ip_bits = IpBits.v4()
            bits = ip_bits.bits
            return Prefix(num, ip_bits, Prefix.new_netmask(num, bits), Prefix32.from_)
        return None


__all__ = ["Prefix32"]
